rust/src/dataset.rs

```diff
@@ -954,6 +955,25 @@ impl Dataset {
         read_manifest_indexes(&self.object_store, &manifest_file, &self.manifest).await
     }
 
+    pub async fn statistics(&self, index_name: String) -> Result<String> {
+        let index_ids = self
+            .load_indices()
+            .await
+            .unwrap()
+            .iter()
+            .filter(|idx| idx.name.eq(&index_name))
```

Comment:
If we are searching by name, we should only get one, so we could use `.find()` instead.

Reply:
Changed.